fastly/block_fastly_service_v1_logging_openstack.go

```diff
@@ -0,0 +1,289 @@
+			"response_condition": ll.ResponseCondition,
+		}
+
+		// Prune any empty values that come from the default string value in structs.
+		for k, v := range nll {
+			if v == "" {
+				delete(nll, k)
+			}
+		}
+
+		lsl = append(lsl, nll)
+	}
+
+	return lsl
+}
+
+func buildCreateOpenstack(openstackMap interface{}, serviceID string, serviceVersion int) *gofastly.CreateOpenstackInput {
+	df := openstackMap.(map[string]interface{})
+
+	return &gofastly.CreateOpenstackInput{
+		Service:           serviceID,
+		Version:           serviceVersion,
+		Name:              gofastly.NullString(df["name"].(string)),
+		URL:               gofastly.NullString(df["url"].(string)),
+		User:              gofastly.NullString(df["user"].(string)),
+		BucketName:        gofastly.NullString(df["bucket_name"].(string)),
+		AccessKey:         gofastly.NullString(df["access_key"].(string)),
+		PublicKey:         gofastly.NullString(df["public_key"].(string)),
+		GzipLevel:         gofastly.Uint(uint(df["gzip_level"].(int))),
+		MessageType:       gofastly.NullString(df["message_type"].(string)),
+		Path:              gofastly.NullString(df["path"].(string)),
+		Period:            gofastly.Uint(uint(df["period"].(int))),
+		TimestampFormat:   gofastly.NullString(df["timestamp_format"].(string)),
+		Format:            gofastly.NullString(df["format"].(string)),
+		FormatVersion:     gofastly.Uint(uint(df["format_version"].(int))),
+		Placement:         gofastly.NullString(df["placement"].(string)),
+		ResponseCondition: gofastly.NullString(df["response_condition"].(string)),
+	}
+}
+
+func buildDeleteOpenstack(openstackMap interface{}, serviceID string, serviceVersion int) *gofastly.DeleteOpenstackInput {
+	df := openstackMap.(map[string]interface{})
+
+	return &gofastly.DeleteOpenstackInput{
+		Service: serviceID,
+		Version: serviceVersion,
+		Name:    df["name"].(string),
+	}
+}
+
+func (h *OpenstackServiceAttributeHandler) Register(s *schema.Resource) error {
+	s.Schema[h.GetKey()] = &schema.Schema{
+		Type:     schema.TypeSet,
+		Optional: true,
+		Elem: &schema.Resource{
+			Schema: map[string]*schema.Schema{
+				// Required fields
+				"name": {
+					Type:        schema.TypeString,
+					Required:    true,
+					Description: "The unique name of the OpenStack logging endpoint.",
+				},
+
+				"url": {
+					Type:        schema.TypeString,
+					Required:    true,
+					Description: "Your OpenStack auth url.",
+				},
+
+				"user": {
+					Type:        schema.TypeString,
+					Required:    true,
+					Description: "The username for your OpenStack account.",
+				},
+
+				"bucket_name": {
+					Type:        schema.TypeString,
+					Required:    true,
+					Description: "The name of your OpenStack container.",
+				},
+
+				"access_key": {
+					Type:        schema.TypeString,
+					Required:    true,
+					Sensitive:   true,
+					Description: "Your OpenStack account access key.",
+				},
+
+				// Optional fields
+				"public_key": {
```

Comment:
Not documented [here](https://developer.fastly.com/reference/api/logging/cloudfiles/), but supported by the API.